Add swab sites to warehouse.site table

truong97 opened the pull request:
Improve ID3C metadata by classifying the swab_site (which we get from the lab manifest) in the warehouse.site table. These swab_site are from mostly from Y2 and Y3 studies.

Additions include:
1) shelter sites
2) skilled nursing facilities sites
3) Snohomish school
4) ROSEY/Yakima School
5) Apple
6) Clinical AFH and Workplace
7) sites for various HCT testing locations
8) Household

truong97 commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Add swab_site metadata for various y3 sites](https://trello.com/c/knzjLbX0/1165-add-swabsite-metadata-for-various-y3-sites)

truong97 commented:
I did deploy-revert to make sure I was doing it correctly on my local database. Below is the revert code if needed:
```
delete from warehouse.site where identifier in ('YouthCare', 'Apple', 'AvalonFederalWayCareCenter', 
'BoeingField', 'BurienNursingAndRehab', 'ClementPlace', 'ClinicalAdultFamilyHomes', 'ClinicalWorkplace',
'ColumbiaLutheranHome', 'CompassBlaineCenter', 'CompassFirstPresbyterian', 'CompassJanAndPetersPlace',
'CompassOttosPlace3rdFloor', 'CompassOttosPlace4thAnd5thFloor', 'CristaRehabAndSkilledNursingCare',
'ExhibitionHall', 'FryeApartments', 'KlineGallandHome', 'Household', 'IssaquahNursingAndRehab',
'JudsonPark', 'JunctionPoint', 'Lazarus', 'LifeCareCenter', 'Marys_PlaceBurien', 'Marys_PlaceNorthSeattle',
'Marys_PlaceNorthshore', 'MarysPlaceRegrade', 'MarysPlaceWhiteCenter', 'Marys_PlaceYesler', 'MissionHealthcare',
'Oaks', 'ParkWestCareCenter', 'ProvidenceMountStVincent', 'RedLionHotel', 'ROOTS', 'ROSEY', 'QueenAnneHealthcare',
'SeattleMedicalPostAcuteCare', 'ShorelineHealthAndRehab', 'SnohomishSchoolDistrict', 'Spruce', 'StaffordHealthcare'); 
```

davereinhart commented:
> I did deploy-revert to make sure I was doing it correctly on my local database. Below is the additional revert code if needed:
> 
> ```
> delete from warehouse.site where identifier in ('YouthCare', 'Apple', 'AvalonFederalWayCareCenter', 
> 'BoeingField', 'BurienNursingAndRehab', 'ClementPlace', 'ClinicalAdultFamilyHomes', 'ClinicalWorkplace',
> 'ColumbiaLutheranHome', 'CompassBlaineCenter', 'CompassFirstPresbyterian', 'CompassJanAndPetersPlace',
> 'CompassOttosPlace3rdFloor', 'CompassOttosPlace4thAnd5thFloor', 'CristaRehabAndSkilledNursingCare',
> 'ExhibitionHall', 'FryeApartments', 'KlineGallandHome', 'Household', 'IssaquahNursingAndRehab',
> 'JudsonPark', 'JunctionPoint', 'Lazarus', 'LifeCareCenter', 'Marys_PlaceBurien', 'Marys_PlaceNorthSeattle',
> 'Marys_PlaceNorthshore', 'MarysPlaceRegrade', 'MarysPlaceWhiteCenter', 'Marys_PlaceYesler', 'MissionHealthcare',
> 'Oaks', 'ParkWestCareCenter', 'ProvidenceMountStVincent', 'RedLionHotel', 'ROOTS', 'ROSEY', 'QueenAnneHealthcare',
> 'SeattleMedicalPostAcuteCare', 'ShorelineHealthAndRehab', 'SnohomishSchoolDistrict', 'Spruce', 'StaffordHealthcare'); 
> ```

You'll want to add that to the bottom of the `revert/warehouse/site/data.sql` file.  After you run `sqitch revert` with the previous tag, your local database should have the exact same records in `warehouse.sites` as production. So you will also have to add some update queries to undo those were updated but were not new.

truong97 commented:
I updated the identifier for UW Ode from `Ode` to `UWOdegaardLibrary`.

Additionally, I compared the CSV files from production `warehouse.sites` table and local `warehouse.sites` table with `diff`. No difference found. When I added a change to local csv and then did `diff`, there was a difference as expected.

kschwabefry commented on schema/deploy/warehouse/site/data.sql:
```diff
@@ -5,44 +5,113 @@ begin;
     values
+        ('Apple',                                   '{"category": "community",  "type": "apple-study", 
+            "sample_origin": "apple"}'),
+        ('AvalonFederalWayCareCenter',              '{"category": "community",  "type": "shelter", 
```
I saw this site mentioned in lists of skilled nursing facilities (e.g. https://seattle-flu-study.slack.com/archives/C0121DE5FEH/p1588968986325500). Does it also operate as a shelter or should the type be switched?

truong97 replied:
@kschwabefry you are correct, avalon should be a snf, not a shelter

kschwabefry commented on schema/deploy/warehouse/site/data.sql:
```diff
@@ -5,44 +5,113 @@ begin;
         ('BirchTreeAcademy',                        '{"category": "community",  "type": "childcare"}'),
+        ('BoeingField',                             '{"category": "community",  "type": "shelter", 
+            "swab_site": "boeing"}'),
+        ('BurienNursingAndRehab',                   '{"category": "community",  "type": "nursingHome", 
```
@truong97 I also saw this group of sites called "skilled nursing facility" or snf, instead of nursing home. Do you have a feel for which term is preferred/ more descriptive?

truong97 replied:
@kschwabefry "skilled nursing facility" is more descriptive (I originally wasn't sure if it was too lengthy). I can change is to that if that is more preferable.

kschwabefry reviewed (commented):
Thanks for adding this in, its great to have this sites updated. I left a couple comments and I'd be interested in your thoughts on them. I know the semantics in the data model aren't super clear, and I think the classifications you've chosen are in line with what we've done previously to the extent possible, which is great.

truong97 commented on schema/deploy/warehouse/site/data.sql:
```diff
@@ -5,44 +5,113 @@ begin;
+            "swab_site": "clement"}'),
+        ('ClinicalAdultFamilyHomes',                '{"category": "clinic",     "type": "adultFamilyHome",
+            "sample_origin": "clinical_afh"}'),
+        ('ClinicalWorkplace',                       '{"category": "clinic",     "type": "clincalWorkplace",
```
@kschwabefry sounds good. Would this apply only to ClinicalWorkplace or ClinicalAdultFamilyHomes as well? I can't seem to remember if the study test only the residents at the AFH and/or also the workers?

kschwabefry replied:
Discussed at standup, but recording here as well: AFH may have tested both residents and staff (similar to childcare studies testing children and staff). The boundary is a little fuzzy but since the main testing is with residents, we'll classify AFH as a SNF rather than a workplace.

kschwabefry reviewed (approved):
Site designations look good to me!

truong97 commented:
Deployed.
